test/integration/httproute_webhook_test.go

```diff
@@ -139,13 +87,199 @@ func TestHTTPRouteValidationWebhook(t *testing.T) {
+						},
+					},
+				},
+				CommonRouteSpec: gatewayv1beta1.CommonRouteSpec{
+					ParentRefs: []gatewayv1beta1.ParentReference{{
+						Namespace: (*gatewayv1beta1.Namespace)(&managedGateway.Namespace),
+						Name:      gatewayv1beta1.ObjectName(managedGateway.Name),
+					}},
+				},
+			},
+		},
+		WantCreateErr:          true,
+		WantCreateErrSubstring: wantCreateErrSubstring,
+	}
+}
+
+func TestHTTPRouteValidationWebhookTraditionalRouter(t *testing.T) {
+	skipTestForNonKindCluster(t)
+	skipTestForRouterFlavors(t, expressions)
+
+	ctx := context.Background()
+	namespace, gatewayClient, managedGateway, unmanagedGateway := setUpEnvForTestingHTTPRouteValidationWebhook(ctx, t)
+	testCases := append(
+		commonHTTPRouteValidationTestCases(managedGateway, unmanagedGateway),
+		invalidRegexInPathTestCase(managedGateway, `HTTPRoute failed schema validation: schema violation (paths.3: invalid regex: '/foo[[[['`),
+		// No test case for invalid regex in header, because Kong Gateway doesn't return any error in such case (it works only for expressions router).
+	)
+	testHTTPRouteValidationWebhook(ctx, t, namespace, gatewayClient, testCases)
+}
+
+func TestHTTPRouteValidationWebhookExpressionsRouter(t *testing.T) {
+	skipTestForNonKindCluster(t)
+	skipTestForRouterFlavors(t, traditional, traditionalCompatible)
+
+	ctx := context.Background()
+	namespace, gatewayClient, managedGateway, unmanagedGateway := setUpEnvForTestingHTTPRouteValidationWebhook(ctx, t)
+	testCases := append(
+		commonHTTPRouteValidationTestCases(managedGateway, unmanagedGateway),
+		invalidRegexInPathTestCase(managedGateway, "regex parse error:\n    ^/foo[[[[\n            ^\nerror: unclosed character class)"),
+		testCaseHTTPRouteValidation{
+			Name: "a httproute with invalid regex for header does not pass validation",
+			Route: &gatewayv1beta1.HTTPRoute{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: uuid.NewString(),
+				},
+				Spec: gatewayv1beta1.HTTPRouteSpec{
+					Hostnames: []gatewayv1beta1.Hostname{"foo.com"},
+					Rules: []gatewayv1beta1.HTTPRouteRule{
+						{
+							Matches: []gatewayv1beta1.HTTPRouteMatch{
+								builder.NewHTTPRouteMatch().WithPathPrefix("/path-6").Build(),
+								builder.NewHTTPRouteMatch().WithHeaderRegex("foo", "bar[[").Build(),
+								builder.NewHTTPRouteMatch().WithPathPrefix("/path-7").Build(),
+							},
+						},
+					},
+					CommonRouteSpec: gatewayv1beta1.CommonRouteSpec{
+						ParentRefs: []gatewayv1beta1.ParentReference{{
+							Namespace: (*gatewayv1beta1.Namespace)(&managedGateway.Namespace),
+							Name:      gatewayv1beta1.ObjectName(managedGateway.Name),
+						}},
+					},
+				},
+			},
+			WantCreateErr:          true,
+			WantCreateErrSubstring: "regex parse error:\n    bar[[\n        ^\nerror: unclosed character class)",
+		},
+	)
+	testHTTPRouteValidationWebhook(ctx, t, namespace, gatewayClient, testCases)
+}
+
+func setUpEnvForTestingHTTPRouteValidationWebhook(ctx context.Context, t *testing.T) (
```

Comment:
One more suggestion for improvement for this would be to add namespace and/or object selectors in `ensureAdmissionRegistration` so that all those webhook tests that run on the same cluster didn't interfere with each other and we get a bit more isolation 😓 

If that's going to be done in a separate PR then 👍 but I'd suggest we create an issue for that.

Reply:
And also that would need to be mentioned in the godoc for this func

Reply:
https://github.com/Kong/kubernetes-ingress-controller/issues/4621